### DISCUSSION TITLE
I am struggling with local preset in selfhosted

### DISCUSSION THREAD

**Discussion by adamgorMSFT** [renovatebot/renovate#15201] [Request Help] [answered]
I am trying to add a local preset to our main renovate.json and I keep getting errors

renovate.json starts off with 
```
{
  "extends": [
    "config:base",
    "group:recommended",
    "src/apps/MyApp/renovate-config"
  ],
...
```

and then that renovate-config.json has something like
```
{
    "packageRules": [
         {
            "description": "Auto self approve all minor and patch updates",
            "matchPaths": ["src/apps/MyApp/package.json"],
            "azureAutoApprove": true,
            "matchUpdateTypes": ["minor", "patch"]
          }
     ]
}
```

For referencing the local preset I've tried
"src/apps/MyApp/renovate-config"
"local>GS/src/apps/MyApp/renovate-config"
"local>GS/src/apps/MyApp:renovate-config"
"local>src/apps/MyApp:renovate-config"
"local>GS//src/apps/MyApp/renovate-config"  // Gave 

GS is the repo name, perhaps I need Xbox/GS there for the org/proj
As we are in https://dev.azure.com/microsoft/Xbox/_git/GS/

I don't need to access this renovate-config in any other repo, except for the one that it applies too.
I just can't seem to choose the right thing to get it to work.  Any tips/suggestions?

Example error

```
Failed to retrieve src/apps/MyApp/renovate-config.json from repo GS(repository=Xbox/GS)
       "err": {
         "message": "Cannot read properties of null (reading 'id')",
         "stack": "TypeError: Cannot read properties of null (reading 'id')\n    at Proxy.getRawFile (/usr/local/lib/node_modules/renovate/dist/modules/platform/azure/index.js:59:23)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at async fetchJSONFile (/usr/local/lib/node_modules/renovate/dist/config/presets/local/common.js:11:15)\n    at async fetchPreset (/usr/local/lib/node_modules/renovate/dist/config/presets/util.js:31:23)\n    at async getPreset (/usr/local/lib/node_modules/renovate/dist/config/presets/index.js:178:24)\n    at async Object.resolveConfigPresets (/usr/local/lib/node_modules/renovate/dist/config/presets/index.js:243:37)\n    at async mergeRenovateConfig (/usr/local/lib/node_modules/renovate/dist/workers/repository/init/merge.js:192:61)\n    at async getRepoConfig (/usr/local/lib/node_modules/renovate/dist/workers/repository/init/config.js:15:14)\n    at async initRepo (/usr/local/lib/node_modules/renovate/dist/workers/repository/init/index.js:27:14)\n    at async Object.renovateRepository (/usr/local/lib/node_modules/renovate/dist/workers/repository/index.js:37:18)\n    at async Object.start (/usr/local/lib/node_modules/renovate/dist/workers/global/index.js:116:13)\n    at async /usr/local/lib/node_modules/renovate/dist/renovate.js:15:24"
       },
       "repo": "GS",
       "fileName": "src/apps/MyApp/renovate-config.json"


DEBUG: Preset fetch error (repository=Xbox/GS)
       "preset": "local>GS//src/apps/MyApp/renovate-config",
       "err": {
         "message": "dep not found",


```

**Comment by rarkins** [comment, marked as answer]
You will definitely need to specify the org name, and I recommend using `local>` as the prefix too. Is `src/apps/MyApp/` a nested path within the repo?

**Reply by adamgorMSFT** [reply]
Thanks! I think I got it working with `local>Xbox/GS//src/apps/MyApp/renovate-config`

And yes `src/apps/MyApp` is the folder path to a file named renovate-config.json

We were going to leverage presets as a way to split out packageRules into different files and appropriate folders, since our monorepo has many different packages that different teams want to control and there was a lot of fuss around having a single large renovate.json at root. And since Renovate doesn't support our scenario the way we wanted, we were going to just self impose all the nested preset configs have appropriate packageMatch per every rule to only impact that particular package rules.
